src/ORBextractor.cc

```diff
@@ -895,6 +949,97 @@ namespace ORB_SLAM3
             computeOrientation(mvImagePyramid[level], allKeypoints[level], umax);
     }
 
+    void ORBextractor::ComputeKeyPointsOctTree(vector<vector<KeyPoint> >& allKeypoints, Mat &depthS, const cv::Mat &K, const cv::Mat &KS, Eigen::Matrix4f &T)
+    {
+        allKeypoints.resize(nlevels);
+
+        const float W = 35;
+
+        for (int level = 0; level < nlevels; ++level)
+        {
+            const int minBorderX = EDGE_THRESHOLD-3;
+            const int minBorderY = minBorderX;
+            const int maxBorderX = mvImagePyramid[level].cols-EDGE_THRESHOLD+3;
+            const int maxBorderY = mvImagePyramid[level].rows-EDGE_THRESHOLD+3;
+
+            const float width = (maxBorderX-minBorderX);
+            const float height = (maxBorderY-minBorderY);
+
+            const int nCols = width/W;
+            const int nRows = height/W;
+            const int wCell = ceil(width/nCols);
+            const int hCell = ceil(height/nRows);
+
+
+            vector<cv::KeyPoint> vToDistributeKeys = vToDistributeKeysCalculate(nRows, minBorderY,
+                                                                                hCell, maxBorderY,
+                                                                                nCols, minBorderX,
+                                                                                wCell, maxBorderX,
+                                                                                level, false);
+            vector<cv::KeyPoint> vToDistributeKeysS = vToDistributeKeysCalculate(nRows, minBorderY,
+                                                                                hCell, maxBorderY,
+                                                                                nCols, minBorderX,
+                                                                                wCell, maxBorderX,
+                                                                                level, true);
+
+            vector<KeyPoint> & keypoints = allKeypoints[level];
+            keypoints.reserve(nfeatures);
+
+            vector<KeyPoint> keypointsTemp = DistributeOctTree(vToDistributeKeys, minBorderX, maxBorderX,
+                                          minBorderY, maxBorderY,mnFeaturesPerLevel[level], level);
+            vector<KeyPoint> keypointsSTemp = DistributeOctTree(vToDistributeKeysS, minBorderX, maxBorderX,
+                                                               minBorderY, maxBorderY,mnFeaturesPerLevel[level], level);
+
+            float x, y, z, u, v;
+            vector<cv::KeyPoint> newKeypointsSTemp;
+            for(auto & i : keypointsSTemp)
```

Comment:
Could be vectorized, should be much faster using Eigen operations in the same manner as with numpy